src/model/docker.ts

```diff
@@ -95,7 +99,173 @@ class Docker {
         ${sshAgent ? '--volume /home/runner/.ssh/known_hosts:/root/.ssh/known_hosts:ro' : ''} \
         ${image}`;
 
-    await exec(command, undefined, { silent });
+        await exec(linuxRunCommand, undefined, { silent });
+        break;
+      }
+      case 'win32': {
+        let unitySerial = '';
+        if (!process.env.UNITY_SERIAL) {
+          //No serial was present so it is a personal license that we need to convert
+          if (!process.env.UNITY_LICENSE) {
+            throw new Error(`Missing Unity License File and no Serial was found. If this
+                             is a personal license, make sure to follow the activation
+                             steps and set the UNITY_LICENSE GitHub secret or enter a Unity
+                             serial number inside the UNITY_SERIAL GitHub secret.`);
+          }
+          unitySerial = this.getSerialFromLicenseFile(process.env.UNITY_LICENSE);
+        } else {
+          unitySerial = process.env.UNITY_SERIAL!;
+        }
+        core.setSecret(unitySerial);
+
+        if (!(process.env.UNITY_EMAIL && process.env.UNITY_PASSWORD)) {
+          throw new Error(`Unity email and password must be set for Windows based builds to
+                           authenticate the license. Make sure to set them inside UNITY_EMAIL
+                           and UNITY_PASSWORD in Github Secrets and pass them into the environment.`);
+        }
+
+        await this.setupWindowsRun(platform);
+
+        this.validateWindowsPlatformRequirements(platform);
```

Comment:
It's fine to do this in a follow-up as well, but this is what the `input` and `buildParameters` models are for. We shouldn't inline some last minute adjustments right before the build command.

The docker model class should only know about how to run the docker image, nothing more.